codemp/qcommon/files.cpp

```diff
@@ -312,6 +312,77 @@ FILE*		missingFiles = NULL;
 #  endif
 #endif
 
+const char *get_filename_ext(const char *filename) {
+	const char *dot = strrchr(filename, '.');
+	if (!dot || dot == filename) return "";
+	return dot + 1;
+}
+
+// We don't want VMs to be able to access the following file extensions
+static char *invalidExtensions[] = {
```

Comment:
an `std::array<const char*>` is nicer to work with; but this could arguably also be an `std::set` (though for a small set like this one, an array is likely more efficient)